quickstart/101-vm-cluster-linux/main.tf

```diff
@@ -0,0 +1,145 @@
+    public_ip_address_id = azurerm_public_ip.test.id
+  }
+}
+
+resource "azurerm_lb_backend_address_pool" "test" {
+  loadbalancer_id = azurerm_lb.test.id
+  name            = "BackEndAddressPool"
+}
+
+resource "azurerm_network_interface" "test" {
+  count               = 2
+  name                = "acctni${count.index}"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+
+  ip_configuration {
+    name                          = "testConfiguration"
+    subnet_id                     = azurerm_subnet.test.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_availability_set" "avset" {
+  name                         = "avset"
+  location                     = azurerm_resource_group.rg.location
+  resource_group_name          = azurerm_resource_group.rg.name
+  platform_fault_domain_count  = 2
+  platform_update_domain_count = 2
+  managed                      = true
+}
+
+resource "random_pet" "azurerm_linux_virtual_machine_name" {
+  prefix = "vm"
+}
+
+resource "random_password" "password" {
+  count       = var.password == null ? 1 : 0
+  length      = 20
+  special     = true
+  min_numeric = 1
+  min_upper   = 1
+  min_lower   = 1
+  min_special = 1
+}
+
+locals {
+  password = try(random_password.password[0].result, var.password)
+}
+
+resource "azurerm_linux_virtual_machine" "test" {
+  count                 = 2
+  name                  = "${random_pet.azurerm_linux_virtual_machine_name.id}${count.index}"
+  location              = azurerm_resource_group.rg.location
+  availability_set_id   = azurerm_availability_set.avset.id
+  resource_group_name   = azurerm_resource_group.rg.name
+  network_interface_ids = [azurerm_network_interface.test[count.index].id]
+  size                  = "Standard_DS1_v2"
+
+  # Uncomment this line to delete the OS disk automatically when deleting the VM
+  # delete_os_disk_on_termination = true
+
+  # Uncomment this line to delete the data disks automatically when deleting the VM
+  # delete_data_disks_on_termination = true
+
+  source_image_reference {
+    publisher = "Canonical"
+    offer     = "UbuntuServer"
+    sku       = "16.04-LTS"
+    version   = "latest"
+  }
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+    name                 = "myosdisk${count.index}"
+  }
+
+  computer_name                   = "hostname"
+  admin_username                  = var.user_name
+  admin_password                  = local.password
+  disable_password_authentication = false
+}
+
+resource "azurerm_managed_disk" "test" {
+  count                = 2
+  name                 = "datadisk_existing_${count.index}"
+  location             = azurerm_resource_group.rg.location
+  resource_group_name  = azurerm_resource_group.rg.name
+  storage_account_type = "Standard_LRS"
+  create_option        = "Empty"
+  disk_size_gb         = "1023"
```

Comment:
Why the size is not `1024` but `1023`?

Reply:
> Thanks @TomArcherMsft, LGTM, only one question remains but the pr is good to merge.

Thanks, @lonegunmanb. I made the change.